Battery_Checker.h

```diff
@@ -0,0 +1,106 @@
+#include <stdio.h>
+#include <assert.h>
+
+#define NOT_OK 0
+#define OK     1
+
+char bmsFactors[3][13] = {"TEMPERATURE", "SOC", "CHARGE RATE"};
+
+enum BMSFactor {
+    TEMPERATURE = 0,
+    SOC,
+    CHARGERATE
+};
+
+enum BMSValueRange {
+ LOW_BREACH = 0,
+ LOW_WARNING,
+ NORMAL_STATE,
+ HIGH_WARNING,
+ HIGH_BREACH
+};
+
+float bmsFactorRanges[15][2]={
+/*Temperature*/
+ /*Low Breach   Low Warning   Normal     High Warning   Hign Breach*/
+   {-30, -1},    {0, 15},    {16, 30},      {31, 45},     {46, 63}, 
```

Comment:
Having hardcoded values could pose problems during extension in future. The question that we need to ask is, if tolerance value is changed, then, will this code remain same? how much changes would be needed?

Reply:
Yes, the current implementation may lead much manual effort while change in tolerance rate.

Now, I made the change to reduce the manual effort.